pilot/pkg/model/push_context.go

```diff
@@ -1036,12 +1035,14 @@ func (ps *PushContext) getExportedDestinationRuleFromNamespace(owningNamespace s
 					if parent = ps.destinationRuleIndex.inheritedByNamespace[clientNamespace]; parent == nil {
 						parent = ps.destinationRuleIndex.inheritedByNamespace[ps.Mesh.RootNamespace]
 					}
-					if child := ps.destinationRuleIndex.exportedByNamespace[owningNamespace].destRule[specificHostname]; child != nil {
-						return ps.inheritDestinationRule(parent, child)
+					if child := ps.destinationRuleIndex.exportedByNamespace[owningNamespace].destRules[specificHostname]; child != nil {
+						return ps.inheritDestinationRule(parent, child[0])
 					}
 					return nil
 				}
-				return ps.destinationRuleIndex.exportedByNamespace[owningNamespace].destRule[specificHostname]
+				if dr, ok := ps.destinationRuleIndex.exportedByNamespace[owningNamespace].destRules[specificHostname]; ok {
+					return dr[0]
```

Comment:
why return just one here?

Reply:
It seems like anywehre we do a `[0]` is a bug?

Reply:
Fixed. Was thinking exported rules will be anyways one, so why iterate. However now have made the code uniform